Source/Core/Core/State.cpp

```diff
@@ -354,6 +306,72 @@ static std::map<double, int> GetSavedStates()
   return m;
 }
 
+static void CompressBufferToFile(const u8* raw_buffer, u64 size, File::IOFile& f)
+{
+  u64 total_bytes_compressed = 0;
+
+  while (true)
+  {
+    u64 bytes_left_to_compress = size - total_bytes_compressed;
+
+    int bytes_to_compress =
+        static_cast<int>(std::min(static_cast<u64>(LZ4_MAX_INPUT_SIZE), bytes_left_to_compress));
+    int compressed_buffer_size = LZ4_compressBound(bytes_to_compress);
+    auto compressed_buffer = std::make_unique<char[]>(compressed_buffer_size);
+    s32 compressed_len = LZ4_compress_default(
+        reinterpret_cast<char*>(const_cast<u8*>(raw_buffer)) + total_bytes_compressed,
+        compressed_buffer.get(), bytes_to_compress, compressed_buffer_size);
```

Comment:
You should be able to just do `reinterpret_cast<const char*>(raw_buffer) here, no?

Reply:
Yes I can do that. I'm not sure why I missed that 😰